Assets/MixedRealityToolkit/Utilities/MixedRealityToolkitSceneChecker.cs

```diff
@@ -0,0 +1,128 @@
+            {
+                // Ignore the check when the scene is explicitly empty.
+                // This includes empty scenes in playmode tests.
+                case NewSceneSetup.EmptyScene:
+                    break;
+
+                case NewSceneSetup.DefaultGameObjects:
+                    CheckMixedRealityToolkitScene();
+                    break;
+            }
+        }
+
+        private static void CheckMixedRealityToolkitScene()
+        {
+            // Create The MR Manager if none exists.
+            if (!MixedRealityToolkit.IsInitialized)
+            {
+                // Search the scene for one, in case we've just hot reloaded the assembly.
+                var managerSearch = Object.FindObjectsOfType<MixedRealityToolkit>();
+
+                if (managerSearch.Length == 0)
+                {
+                    HideNoActiveToolkitWarning = SessionState.GetBool(HideNoActiveToolkitWarningKey, false);
+                    if (!HideNoActiveToolkitWarning)
+                    {
+                        NoActiveToolkitWarning.OpenWindow(null);
+                    }
+                    return; 
+                }
+            }
+        }
+
+        internal class NoActiveToolkitWarning : EditorWindow
+        {
+            private static NoActiveToolkitWarning activeWindow;
+            [SerializeField]
+            public MixedRealityToolkitConfigurationProfile configurationProfile;
+            private bool hideWarning = false;
+
+            public static void OpenWindow(MixedRealityToolkitConfigurationProfile configurationProfile)
+            {
+                // If we already have an active window, bail
+                if (activeWindow != null)
+                    return;
+
+                activeWindow = EditorWindow.GetWindow<NoActiveToolkitWarning>();
+                activeWindow.configurationProfile = configurationProfile;
+                activeWindow.maxSize = new Vector2(400, 140);
+                activeWindow.minSize = new Vector2(400, 140);
+                activeWindow.titleContent = new GUIContent("No Active Toolkit Found");
+
+                activeWindow.ShowPopup(); 
+            }
+
+            private void OnGUI()
+            {
+                EditorGUILayout.HelpBox("There is no active Mixed Reality Toolkit in your scene. Would you like to create one now?", MessageType.Warning);
+
+                hideWarning = EditorGUILayout.Toggle("Don't show this again", hideWarning);
+
+                configurationProfile = (MixedRealityToolkitConfigurationProfile)EditorGUILayout.ObjectField(configurationProfile, typeof(MixedRealityToolkitConfigurationProfile), false);
+                if (configurationProfile == null)
+                {
+                    EditorGUILayout.HelpBox("Select a configuration profile.", MessageType.Info);
+                }
+
+                EditorGUILayout.BeginHorizontal();
+                GUI.enabled = (configurationProfile != null);
```

Comment:
>GUI.enabled = (configurationProfile != null); [](start = 16, length = 45)

Reading the code but without great background knowledge on building unity editor UI, it seems a bit odd that we're first setting the enabled bit here conditionally, and then blowing it away after the if statement by always setting it to true on line 115. This question is more for my own understanding, but is this designed to disable the Yes button conditionally (i.e. it affects anything after it's been changed), but the No button will be left enabled in all cases because GUI.enabled is true after that point?

Reply:
Correct, this disables the "Yes" button unless a profile has been selected (the `GUI.enabled` state applies to all statements after the change). Could be removed if that seems too restrictive, but since the purpose is to end up with a valid MRTK object i think it's helpful. I've tried selecting the default profile, but we don't actually have any definition for that.